### PR TITLE
feat!: add preprocessing step in get_data

### PR DESCRIPTION
**Pull request by jolars**
This PR adds a data preprocessing step for all (current) data sets. Sparse datasets are scaled by the maximum absolute value and dense datasets are centered my mean and scaled by standard deviation.

### CODE REVIEW

File: datasets/leukemia.py [hunk 1]
```diff
@@ -16,8 +18,10 @@ def get_data(self):
...
-        X = X.to_numpy()
-        y = LabelBinarizer().fit_transform(y)[:, 0].astype(X.dtype)
+        self.X, self.y = fetch_openml("leukemia", return_X_y=True)
+        self.X = self.X.to_numpy()
+        self.y = LabelBinarizer().fit_transform(self.y)[:, 0].astype(self.X.dtype)
+
+        self.X, self.y = preprocess_data(self.X, self.y)
 
-        return dict(X=X, y=y)
+        return dict(X=self.X, y=self.y)
```
Comment (tomMoral) [inline datasets/leukemia.py]: Why do you put this in the object? I it is not necessary, I would advise against it, so it does not mess up with hashing and serialization (it should no but we are never too careful :) )

**Review by tomMoral — COMMENTED** [review]
Nice! I think we should document this use case in the benchopt doc.
Maybe we could start a sort of "gallery" where we put pointer on simple code that use a certain feature, such as this one for the `utils` folder?

**Comment by jolars** [comment]
> Nice! I think we should document this use case in the benchopt doc.
> Maybe we could start a sort of "gallery" where we put pointer on simple code that use a certain feature, such as this one for the `utils` folder?

That's not a bad idea. Maybe this doesn't quite fit the bill, but I guess we could consider having a `contrib` folder in the repo too?

> Why do you put this in the object? I it is not necessary, I would advise against it, so it does not mess up with hashing and serialization (it should no but we are never too careful :) )

I don't know actually... I copied it from somewhere else :) I'll fix it!

Thanks for the review(s)!

**Comment by tomMoral** [comment]
Not sure what is the meaning of a `contrib` folder?

**Comment by jolars** [comment]
Well, in my experience the contrib folder is a standard way of including useful code that's been contributed by someone but isn't deemed important enough to include as main functionality in the package (and is not maintained by the project). That seems to apply quite well here, no?

File: datasets/leukemia.py (hunk 1 above)
Comment (jolars) [reply datasets/leukemia.py]: Thanks for pointing it out, I fixed it in ebb4ba4.

**Pull request merged**